src/agent/src/linux_abi.rs

```diff
@@ -15,9 +17,30 @@ pub const SYSFS_PCI_BUS_RESCAN_FILE: &str = "/sys/bus/pci/rescan";
     target_arch = "x86_64",
     target_arch = "x86"
 ))]
-pub const PCI_ROOT_BUS_PATH: &str = "/devices/pci0000:00";
+pub fn create_pci_root_bus_path() -> String {
+    String::from("/devices/pci0000:00")
+}
+
 #[cfg(target_arch = "aarch64")]
-pub const PCI_ROOT_BUS_PATH: &str = "/devices/platform/4010000000.pcie/pci0000:00";
+pub fn create_pci_root_bus_path() -> String {
+    let mut sysfs_dir = String::from(SYSFS_DIR);
+
+    let mut start_root_bus_path = String::from("/devices/platform");
+    let end_root_bus_path = String::from("/pci0000:00");
+
+    sysfs_dir.push_str(&start_root_bus_path);
+
+    let files = fs::read_to_string(sysfs_dir);
```

Comment:
read_to_string fn return an Result<>, I think you should deal with the Result value instead of using it directly.

Reply:
thanks @lifupan -, I change a lot this version.